rfcs/2021-05-17-open-connection.md

```diff
@@ -0,0 +1,50 @@
+# Open connections
+
+## Problem
+
+This proposal describes how to create invitations that can be used multiple times. 
+
+It can be used for:
+- an open invitation to join a group.
+- an open invitation to connect to a person - e.g. QR code/invitation link on a person's website, or passed from one person to another.
+- part of the solution for public DNS-based addresses (when a directory server would map address in some domain name#example.tld to an open invitation).
+
+## Solution
+
+A minimal change to an SMP protocol - a dedicated unsecured SMP queue is used to receive invitations to connect: instead of the initial KEY message with the KEY to secure the queue, an additional INV message will be used to send an invitation to the queue of the sender, effectively using an unsecured SMP queue as an out-of-band channel for creating SMP queue.
+
+Additional commands in SMP agent protocol:
+
+`conn_alias? OPEN` (or `PUB`, `NEWPUB`, tbc) - to create an "open"/"public" queue, the response is an invitation in a different format (TBC):
+  - should allow multiple servers (probably the original invitation should be extended to support it)
+  - should have a marker to indicate it's an open/public queue (probably the original invitation should be extended to include an invitation type).
+
+e.g. `smp:<queue_type>::<server1>/<queue_id1>,<server2>/<queue_id2>::<key1>`
```

Comment:
rename `<queue_type>` -> `<invitation_type>`? I understand it's for substitution anyway, but if we add it to protocol I think it's best renamed, otherwise it's ambiguous